contracts/AgentMech.sol

```diff
@@ -1,8 +1,35 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.21;
+pragma solidity ^0.8.25;
 
 import {ERC721Mech} from "../lib/gnosis-mech/contracts/ERC721Mech.sol";
 
+// Mech delivery info struct
+struct MechDelivery {
+    // Priority mech address
+    address priorityMech;
+    // Delivery mech address
+    address deliveryMech;
+    // Account address sending the request
+    address account;
+    // Response timeout window
+    uint32 responseTimeout;
+}
+
+// Mech Marketplace interface
+interface IMechMarketplace {
+    /// @dev Delivers a request.
+    /// @param requestId Request id.
+    /// @param requestIdWithNonce Request id with nonce.
+    /// @param requestData Self-descriptive opaque data-blob.
+    function deliver(uint256 requestId, uint256 requestIdWithNonce, bytes memory requestData) external;
```

Comment:
do we still need both requestId and requestIdWithNonce?

Reply:
Seems like not, we just need a `requestIdWithNonce`, but for a possible compatibility we left as is. @0xArdi could you please comment on whether we need a pure `requestId` value?

Reply:
IIRC, we added this in because otherwise if two requests use the same data, only one of them will ever get an answer.

Reply:
@kupermind i agree with your comment. Should we just call it requestId but have it calculated in the way that requestIdWithNonce is?

Reply:
@0xArdi yes that's what I was thinking, thanks!